platforms/avr/drivers/uart.h

```diff
@@ -31,6 +31,7 @@ void uart_init(uint32_t baud);
 void uart_write(uint8_t data);
 
 uint8_t uart_read(void);
+int16_t uart_read2(void); // non-blocking
```

Comment:
The addition here would be invalid, as it creates inconsistencies in what is meant to be a generic API.

The net result is your change would only successfully compile on AVR.

Reply:
OK, thank you for your reviews.
I will close this PR and re-refine and re-think strategy to support Adafruit BLE UART friends.

Reply:
This function can also be implemented using the existing APIs:
```c
int16_t uart_read_nonblock() {
    if (uart_available()) {
        return uart_read();
    } else {
        return -1;
    }
}
```

Reply:
wow, it works! thank you for your advice

Reply:
I removed unecessary changes about generic UART driver's code and re-pushed. thank you.